python/pyspark/sql/connect/dataframe.py

```diff
@@ -137,7 +137,13 @@ def isEmpty(self) -> bool:
         return len(self.take(1)) == 0
 
     def select(self, *cols: "ColumnOrName") -> "DataFrame":
-        return DataFrame.withPlan(plan.Project(self._plan, *cols), session=self._session)
+        sql_expr = []
+        for element in cols:
+            if isinstance(element, str):
+                sql_expr.append(sql_expression(element))
```

Comment:
Do we have a proto message for `UnresolvedAttribute`?

Reply:
I think the string here must be column name, otherwise people should call `selectExpr`

Reply:
This is something I am not sure for Python (PySpark):

`spark.range(10).groupBy("id").sum("id").select("sum(id)")` this works in PySpark.

Reply:
This is working example makes sense then it sounds like PySpark extends the power of `select` compared to Scala DataFrame? cc @HyukjinKwon @zhengruifeng

Reply:
`spark.range(10).groupBy("id").sum("id")` creates a data frame with a column name called `sum(id)`

Reply:
I think using sql expression is probably a bit too aggressive here. I'm currently experimenting with surrounding this with `quoteIfNeeded`.

Reply:
Unfortunately that does not work either:

```
s2.read.table("martin.trips").groupBy("martin.trips.pickup_zip").sum("fare_amount").select("sum(fare_amount)").show()
```

Now the `sum()` is working but `martin.trips.pickup_zip` becomes a quoted identifier as well.

```
	details = "[UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `martin.trips.pickup_zip` cannot be resolved. Did you mean one of the following? [`spark_catalog`.`martin`.`trips`.`pickup_zip`, `spark_catalog`.`martin`.`trips`.`fare_amount`];
'Project [unresolvedalias('sum(fare_amount), None)]
+- 'Aggregate ['`martin.trips.pickup_zip`], ['`martin.trips.pickup_zip`, sum(fare_amount#19) AS sum(fare_amount)#60]
   +- SubqueryAlias spark_catalog.martin.trips
      +- Relation spark_catalog.martin.trips[pickup_zip#18,fare_amount#19] parquet
"
```

Reply:
@amaliujia This approach does not work either. If I apply your approach to my example, it fails as well:

```
s2.read.table("martin.trips").groupBy("martin.trips.pickup_zip").sum("fare_amount").select("sum(fare_amount)").show()
```

Error

```
[UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `fare_amount` cannot be resolved. Did you mean one of the following? [`sum(fare_amount)`, `spark_catalog`.`martin`.`trips`.`pickup_zip`]; line 1 pos 4;
'Project [unresolvedalias('sum('fare_amount), None)]
+- Aggregate [pickup_zip#0], [pickup_zip#0, sum(fare_amount#1) AS sum(fare_amount)#42]
   +- SubqueryAlias spark_catalog.martin.trips
      +- Relation spark_catalog.martin.trips[pickup_zip#0,fare_amount#1] parquet
```